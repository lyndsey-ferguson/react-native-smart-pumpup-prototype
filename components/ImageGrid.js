/**
 ImageGrid.js
 Smart Components

 React Native app created by Lyndsey on 10/7/17.
 Copyright Â© 2017 Lyndsey Ferguson Apps. All rights reserved.
*/

import React, {
  Component
} from 'react';
import {
  Dimensions,
  Image,
  StyleSheet,
  View
} from 'react-native';
import Grid from 'react-native-grid-component';
import deepEqual from 'deep-equal';

export default class ImageGrid extends React.Component {
  renderImageGridItem(imageItem, i) {
    const { imagesList } = this.props;

    const imageUri = imagesList.length > 0 ? imagesList[0].uri : 'https://regmedia.co.uk/2017/06/02/fireball.jpg';

    return (
      <Image style={[styles.item, { backgroundColor: 'white' }]}
       resizeMode='contain'
       source={{uri: imageItem.uri}}
       defaultSource={require('../res/no_avatar.png')}
       key={imageItem.key}/>
    );
  }
  render() {
    const { imagesList } = this.props;

    return (
      <Grid style={styles.list}
      renderItem={this.renderImageGridItem.bind(this)}
      data={imagesList}
      itemsPerRow={3}
      itemHasChanged={ (d1, d2) => d1 !== d2 && !deepEqual(d1, d2)}
      />
    )
  }
}

const { width: windowWidth } = Dimensions.get('window');

const styles = StyleSheet.create({
  item: {
    flex: 1,
    height: (windowWidth - 6) / 3,
    margin: 1
  },
  list: {
    flex: 1
  },
});
