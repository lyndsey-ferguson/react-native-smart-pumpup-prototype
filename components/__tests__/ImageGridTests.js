import React from 'react';
import ImageGrid from '../ImageGrid';
import renderer from 'react-test-renderer';

it('renders correctly without any images', () => {
  const tree = renderer.create(
    <ImageGrid imagesList={[]}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with 6 images', () => {
  const tree = renderer.create(
    <ImageGrid imagesList={[
      { uri: 'https://www.example.com/image1.png', key: '1'},
      { uri: 'https://www.example.com/image2.png', key: '2'},
      { uri: 'https://www.example.com/image3.png', key: '3'},
      { uri: 'https://www.example.com/image4.png', key: '4'},
      { uri: 'https://www.example.com/image5.png', key: '5'},
      { uri: 'https://www.example.com/image6.png', key: '6'},
    ]}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
