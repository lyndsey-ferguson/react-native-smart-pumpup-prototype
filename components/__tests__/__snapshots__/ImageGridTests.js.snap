// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with 6 images 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "key": "1",
          "uri": "https://www.example.com/image1.png",
        },
        Object {
          "key": "2",
          "uri": "https://www.example.com/image2.png",
        },
        Object {
          "key": "3",
          "uri": "https://www.example.com/image3.png",
        },
        Object {
          "key": "4",
          "uri": "https://www.example.com/image4.png",
        },
        Object {
          "key": "5",
          "uri": "https://www.example.com/image5.png",
        },
        Object {
          "key": "6",
          "uri": "https://www.example.com/image6.png",
        },
      ]
    }
    dataSource={
      ListViewDataSource {
        "items": 2,
      }
    }
    enableEmptySections={true}
    itemHasChanged={[Function]}
    itemsPerRow={3}
    onEndReached={[Function]}
    onEndReachedThreshold={1334}
    refreshControl={null}
    renderFooter={[Function]}
    renderItem={[Function]}
    renderPlaceholder={null}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    renderSectionHeader={[Function]}
    sections={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          }
        }
      >
        <Image
          defaultSource={1}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://www.example.com/image1.png",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": 248,
                "margin": 1,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        />
        <Image
          defaultSource={1}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://www.example.com/image2.png",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": 248,
                "margin": 1,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        />
        <Image
          defaultSource={1}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://www.example.com/image3.png",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": 248,
                "margin": 1,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          }
        }
      >
        <Image
          defaultSource={1}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://www.example.com/image4.png",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": 248,
                "margin": 1,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        />
        <Image
          defaultSource={1}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://www.example.com/image5.png",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": 248,
                "margin": 1,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        />
        <Image
          defaultSource={1}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://www.example.com/image6.png",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": 248,
                "margin": 1,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly without any images 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    enableEmptySections={true}
    itemHasChanged={[Function]}
    itemsPerRow={3}
    onEndReached={[Function]}
    onEndReachedThreshold={1334}
    refreshControl={null}
    renderFooter={[Function]}
    renderItem={[Function]}
    renderPlaceholder={null}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    renderSectionHeader={[Function]}
    sections={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
